integrated_channels/sap_success_factors/client.py

```diff
@@ -235,3 +239,87 @@ def _call_post_with_session(self, url, payload):
             self._create_session()
         response = self.session.post(url, data=payload)
         return response.status_code, response.text
+
+    def get_inactive_sap_learners(self):
+        """
+        Make a GET request using the session object to a SuccessFactors endpoint for inactive learners.
+
+        Example:
+            sap_search_student_url: "/learning/odatav4/searchStudent/v1/Students?$filter=criteria/isActive eq False"
+            SAP API response: {
+                u'@odata.metadataEtag': u'W/"17090d86-20fa-49c8-8de0-de1d308c8b55"',
+                u'value': [
+                    {
+                        u'studentID': u'admint6',
+                        u'firstName': u'Compensation',
+                        u'profileStatus': u'ACTIVE',
+                        u'lastName': u'Admin',
+                        u'personExternalID': u'admint6',
+                        u'emailAddr': u'',
+                        u'notActive': u'yes',
+                        u'termDate': None
+                    },
+                    {
+                        u'studentID': u'adminsap1',
+                        u'firstName': u'SAP1',
+                        u'profileStatus': u'ACTIVE',
+                        u'lastName': u'Administrator',
+                        u'personExternalID': u'adminsap1',
+                        u'emailAddr': u'',
+                        u'notActive': u'yes',
+                        u'termDate': None
+                    }
+                ]
+            }
+
+        Returns: List of inactive learners data
+        [
+            {
+                u'studentID': u'admint6',
+                u'firstName': u'Compensation',
+                u'profileStatus': u'ACTIVE',
+                u'lastName': u'Admin',
+                u'personExternalID': u'admint6',
+                u'emailAddr': u'',
+                u'notActive': u'yes',
+                u'termDate': None
+            },
+            {
+                u'studentID': u'adminsap1',
+                u'firstName': u'SAP1',
+                u'profileStatus': u'ACTIVE',
+                u'lastName': u'Administrator',
+                u'personExternalID': u'adminsap1',
+                u'emailAddr': u'',
+                u'notActive': u'yes',
+                u'termDate': None
+            }
+        ]
+        """
+        now = datetime.datetime.utcnow()
+        if now >= self.expires_at:
+            # Create a new session with a valid token
+            self.session.close()
+            self._create_session()
+
+        sap_search_student_url = '{sapsf_base_url}/{search_students_path}?$filter={search_filter}'.format(
+            sapsf_base_url=self.enterprise_configuration.sapsf_base_url.rstrip('/'),
+            search_students_path=self.global_sap_config.search_student_api_path.rstrip('/'),
+            search_filter='criteria/isActive eq False&'
+                          '$select=studentID,personExternalID,firstName,lastName,'
```

Comment:
We should be sure to select only those fields which are necessary to correctly perform the inactivity check -- it appears that only the studentID is being used at present.